compiler/jsgen.nim

```diff
@@ -2043,7 +2043,11 @@ proc genMagic(p: PProc, n: PNode, r: var TCompRes) =
   of mNew, mNewFinalize: genNew(p, n)
```

Comment:
add changelog entry in `## Compiler changes` mentioning this change

Reply:
though it should, but I think the internal of JS array is not documented?

Reply:
it's not, but this is still an improvement worth mentioning for whoever looks at changelog